ngraph/core/include/ngraph/op/util/sub_graph_base.hpp

```diff
@@ -15,39 +15,16 @@ namespace ngraph
         {
             /// \brief Abstract base class for sub-graph based ops, i.e ops that have sub-graph
             ///
-            class NGRAPH_API SubGraphOp : public Op
+            class NGRAPH_API SubGraphOp : public MultiSubGraphOp
             {
             public:
                 NGRAPH_RTTI_DECLARATION;
-                /// \brief Describes a connection between a SubGraphOp input and the body.
-                class InputDescription
-                {
-                protected:
-                    ///
-                    /// \brief      Constructs a new instance.
-                    ///
-                    /// \param      input_index           Position of the SubGraphOp input
-                    /// \param      body_parameter_index  Body parameter to receive input
-                    ///
-                    InputDescription(uint64_t input_index, uint64_t body_parameter_index);
-                    InputDescription() = default;
-
-                public:
-                    using type_info_t = DiscreteTypeInfo;
-                    virtual ~InputDescription() = default;
-                    virtual std::shared_ptr<InputDescription> copy() const = 0;
-
-                    virtual const type_info_t& get_type_info() const = 0;
-
-                    uint64_t m_input_index{0};
-                    uint64_t m_body_parameter_index{0};
-                };
 
                 ///
                 /// \brief      Describes a body input formed from slices of an input to
                 ///             SubGraphOp.
                 ///
-                class NGRAPH_API SliceInputDescription : public InputDescription
+                class NGRAPH_API SliceInputDescription : public MultiSubGraphOp::InputDescription
```

Comment:
Can we move all these classes (Slice, Merged, Concat) to MultiSubgraph?

Reply:
I think no, they are specific classes for SubGraphOp and its derived classes.

Reply:
I don't see any problems with it, the classes (Slice, Merged, Concat) will be used only where necessary.